.github/workflows/synth.yml

```diff
@@ -0,0 +1,29 @@
+name: Synth
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - edited
+      - synchronize
+      - labeled
+      - unlabeled
+      - reopened
+  workflow_dispatch: {}
+
+jobs:
+  synth:
+    runs-on: ubuntu-latest
+    container:
+      image: docker.mirror.hashicorp.services/hashicorp/jsii-terraform
+    steps:
+      - name: Checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+      - name: ensure correct user
+        run: chown -R root /__w/cdktf-integration-serverless-python-gcp-example/
+      - name: Install dependencies
+        run: yarn && pipenv install
+      - name: Synth
+        run: yarn synth
```

Comment:
🤔 I understand why you did it this way but it feels a little weird to me to suddenly add Node.js as a dependency for this Python project. I wonder if you could instead install the cdktf cli via Homebrew (see here: https://developer.hashicorp.com/terraform/tutorials/cdktf/cdktf-install#install-cdktf) and then run `cdktf synth` instead. Everything else will still stay the same, but you can get rid of the `package.json` and `yarn.lock` files (and the `node_modules` in .gitignore).

Reply:
Good point– people might find the presence of a `package.json` a bit confusing 😅.

Gave the homebrew route a try, and it was a bit of a pain since both `docker` and `pipenv` don't come pre-loaded on the `macos` runner. Instead I just put all the yarn stuff into the workflow (yarn project creation and adding cdktf-cli as a global dependency). Accomplishes having access to the cdktf-cli without introducing the extra noise of yarn stuff being in the repo directly.